### PR TITLE
Harmonize spacing for template headers.

### LINKED ISSUE #53: Inconsistent spacing between "template" and "<"
From an email to cxxeditor@:

```
template<typename T>
```

or

```
template <typename T>
```

The standard as-is has 1494 instances of 'template<' that I can find trivially, and 2053 instances of 'template <'.

### PR DESCRIPTION
**Pull request by jensmaurer**
Use `template<class T>`.
Also remove space between two closing template brackets in [lib].
Leave [temp] unchanged, intentionally showing a variety of styles.

Fixes #53.

### CODE REVIEW

**Comment by jensmaurer** [comment]
Rebased.

**Comment by jwakely** [comment]
This looks good to me.

**Comment by tkoeppe** [comment]
All checks have failed!

**Comment by jensmaurer** [comment]
There's only one check, and it's the check-whitespace check, apparently preventing trailing whitespace.

**Comment by jensmaurer** [comment]
Let's try again.

**Comment by tkoeppe** [comment]
I imagine this PR would need some rebasing, but I'm not sure if we want a monolithic commit here. @zygoloid, how would you like to approach this?

File: source/algorithms.tex [hunk 1]
```diff
@@ -555,13 +555,13 @@
                                            BidirectionalIterator first,
                                            BidirectionalIterator last,
                                            Predicate pred);
-  template <class InputIterator, class OutputIterator1,
+  template<class InputIterator, class OutputIterator1,
             class OutputIterator2, class Predicate>
```
Comment (zygoloid) [inline source/algorithms.tex]: This line and those like it need to have their alignment fixed.

**Comment by zygoloid** [comment]
The difference between one commit per file and one commit in total seems inconsequential to me from a review perspective. I'm fine with letting the change author make that call.

**Comment by tkoeppe** [comment]
Rebase?

**Comment by jensmaurer** [comment]
This needs more than a rebase. I also need to review all the alignments (see @zygoloid's comment).

File: source/algorithms.tex (hunk 1 above)
Comment (jensmaurer) [reply source/algorithms.tex]: Agreed. I've scriptified the search and found a few pre-existing misalignments, too. Fixed.

**Comment by jensmaurer** [comment]
Rebased (in fact, re-done) and misalignments fixed.

**Pull request merged**